vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -381,12 +381,13 @@ import { isFocusable } from '@vaadin/grid/src/vaadin-grid-active-item-mixin.js';
               ensureSubCacheQueue = [];
               callback(cache[parentUniqueKey][page], cache[parentUniqueKey].size);
 
-              // Flush after the callback to have the grid rows up-to-date
-              updateAllGridRowsInDomBasedOnCache();
+              // Update effective size
+              grid._cache.updateSize();
+              grid._effectiveSize = grid._cache.effectiveSize;
```

Comment:
`updateAllGridRowsInDomBasedOnCache` consists of the lines added here and an explicit call to update all the grid's rendered rows. Since we only require the size update here, we can skip the unnecessary (and heavy) content update. A manual update request is made at the end of the block.

Reply:
Ideally, these size updates shouldn't be needed because there is `_debouncerApplyCachedData` in the web component that does exactly the same with a 0ms delay.